src/lib/collision_prevention/CollisionPrevention.cpp

```diff
@@ -394,24 +394,24 @@ CollisionPrevention::_addDistanceSensorData(distance_sensor_s &distance_sensor,
 	// discard values below min range
 	if (distance_reading > distance_sensor.min_distance) {
 
-		// get drone attitude in rad
-		Eulerf euler_vehicle(vehicle_attitude);
-		float roll  = euler_vehicle.phi();
-		float pitch = euler_vehicle.theta();
-
-
 		float sensor_yaw_body_rad = _sensorOrientationToYawOffset(distance_sensor, _obstacle_map_body_frame.angle_offset);
 		float sensor_yaw_body_deg = math::degrees(wrap_2pi(sensor_yaw_body_rad));
 
 		// calculate the field of view boundary bin indices
 		int lower_bound = (int)round((sensor_yaw_body_deg  - math::degrees(distance_sensor.h_fov / 2.0f)) / BIN_SIZE);
 		int upper_bound = (int)round((sensor_yaw_body_deg  + math::degrees(distance_sensor.h_fov / 2.0f)) / BIN_SIZE);
 
-		Vector2f sensor_unit_vector = Vector2f(cosf(sensor_yaw_body_rad), sinf(sensor_yaw_body_rad)).unit_or_zero();
-		Vector2f rotated_sensor_unit_vector = _rotatePointByPitchAndRoll(sensor_unit_vector, pitch, roll);
+		Quatf q_vehicle_attitude = Quatf(vehicle_attitude);
+		Quatf q_sensor           = Quatf(Eulerf(0.0f, 0.0f, sensor_yaw_body_rad));
```

Comment:
To reduce the amount of operations, you can even shortcut this by doing: 
`Quatf q_sensor(cosf(sensor_yaw_body_rad / 2.f), 0.f, 0.f, sinf(sensor_yaw_body_rad / 2.f));`